src/modules/Dashboard/components/ParticipantDropdown/ParticipantDropdown.hooks.ts

```diff
@@ -0,0 +1,271 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { DEFAULT_ROWS_PER_PAGE, Roles } from 'shared/consts';
+import { ParticipantsData } from 'modules/Dashboard/features/Participants';
+import { getWorkspaceManagersApi, getWorkspaceRespondentsApi } from 'api';
+import { useAsync } from 'shared/hooks';
+import { Manager, Respondent } from 'modules/Dashboard/types';
+import { auth, workspaces } from 'redux/modules';
+
+import {
+  ParticipantDropdownOption,
+  UseParticipantDropdownProps,
+  AnyTeamSearchType,
+  FullTeamSearchType,
+} from './ParticipantDropdown.types';
+import { participantToOption } from './ParticipantDropdown.utils';
+
+const ALLOWED_TEAM_MEMBER_ROLES: readonly Roles[] = [
+  Roles.SuperAdmin,
+  Roles.Owner,
+  Roles.Manager,
+] as const;
+
+export const useParticipantDropdown = ({
```

Comment:
Unrelated to this task, but showing "all participants" displays anonymous participants as targets for Assign (both target and respondent subjects)...which should probably not be possible.

![CleanShot 2024-08-06 at 16 19 23](https://github.com/user-attachments/assets/20873108-e272-46d7-8606-f974146c0c63)